app/serializers/framework_flag_serializer.rb

```diff
@@ -5,7 +5,7 @@ class FrameworkFlagSerializer
 
   set_type :framework_flags
 
-  belongs_to :question, serializer: FrameworkQuestionSerializer, &:framework_question
+  belongs_to :question, serializer: FrameworkQuestionSerializer, object_method_name: 'framework_question', id_method_name: 'framework_question_id'
```

Comment:
I never quite understood this approach vs just calling the method as a block argument. Did you find this improves the DB queries? Also would this work with symbols for the method names - might look slightly cleaner if so.

Reply:
I was hoping to avoid this, but it seems if we include something such as a map for the real name of relationship i.e. `&:framework_question` it overwrites the default serializer methods, and ignores everything loaded into memory and retrieves it again causing n+1 queries. By passing in explicitly the object names/id methods we allow the JSON serializer to lookup those values normally. I'll try to see if it works with symbols

Reply:
Interesting... then I wonder if this approach could be adopted a bit more widely and it may mitigate the redundant reloading you saw in `create` controller method above 🤔

Reply:
Discussed in person - using the block syntax does indeed cause a torrent of N+1 queries so this is a better approach to adopt, but it only seems to help with `belongs_to` relationships and does not improve `has_many`. Given that we have a mixture of both relationships across the serializer graph we can't fix the N+1 loading everywhere like this so need to rely on eager loading the relevant active record relationships where appropriate. Moving forward, we can adopt this pattern for other `belongs_to` serializer relationships that need to present a different name in the serialized output.